subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaExecutableUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.internal.deprecation.DeprecationLogger;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+
+public class JavaExecutableUtils {
+
+    public static File validateExecutable(String executable) {
+        File executableFile = new File(executable);
+        if (!executableFile.isAbsolute()) {
+            DeprecationLogger.deprecateBehaviour("Configuring a Java executable via a relative path.")
+                    .withContext("Resolving relative file paths might yield unexpected results.")
```

Comment:
:lipstick: I would also add an advice, something like:
```suggestion
                    .withContext("Resolving relative file paths might yield unexpected results.")
                    .withAdvice("Configure an absolute path to a Java executable instead.")
```